lib/flak/assist.go

```diff
@@ -39,3 +40,20 @@ func GetTempDir(dirs ...string) string {
 
 	return folder
 }
+
+// Runs a function with a tempdir, cleaning up afterward.
+func WithTempDir(f func(string), dirs ...string) {
+
+	// If IOError occurs while creating tempdir, final dir was not created to clean up.
+	dir := GetTempDir(dirs...)
+
+	try.Do(func() {
+		f(dir)
+	}).Finally(func() {
+		err := os.RemoveAll(dir)
+		if err != nil {
+			// TODO: we don't want to panic here, more like a debug log entry, "failed to remove tempdir."
+			// Can accomplish once we add logging.
```

Comment:
When in doubt, let's panic anyway?  Def should be a (loud) log statement; let's not swallow until then because it dramatically increases the odds we'll forget about it.

(I've spent a lot of time debugging in golang projects only to end in discovering a `// TODO: handle error` :tada:, druther avoid it out of the gate)